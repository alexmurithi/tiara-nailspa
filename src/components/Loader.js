import React from 'react'
import { CircularProgress, Box } from '@mui/material'
import { makeStyles } from 'tss-react/mui'

const useStyles = makeStyles()(theme => {
  return {
    loaderWrapper: {
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
      height: '100vh',
    },
    loader: {
      margin: theme.spacing(2),
      color: theme.palette.primary.main,
    },
  }
})

const Loader = () => {
  const { classes } = useStyles()

  return (
    <Box className={classes.loaderWrapper}>
      <CircularProgress className={classes.loader} />
    </Box>
  )
}

export default Loader
