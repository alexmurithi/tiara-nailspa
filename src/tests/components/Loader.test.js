/* eslint-disable no-undef */
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from 'react-test-renderer'
import Loader from '@components/Loader'
import { ThemeProvider } from '@mui/material'
import theme from '@themes/Default'

describe('<Loader/>', () => {
  it('renders loader without crashing', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={theme}>
          <Loader />
        </ThemeProvider>,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
