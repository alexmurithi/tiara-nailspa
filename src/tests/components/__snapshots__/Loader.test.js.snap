// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader/> renders loader without crashing 1`] = `
<div
  className="MuiBox-root css-9g93ls-loaderWrapper"
>
  <span
    className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18hv61u-MuiCircularProgress-root-loader"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </span>
</div>
`;
